EthereumKit/Networking/Request/JSONRPC.swift

```diff
@@ -80,4 +80,45 @@ public final class JSONRPC {
             return Response(id: transactionID)
         }
     }
+    
+    public struct Call: JSONRPCKit.Request {
+        public typealias Response = String
+        
+        public let from: Address?
+        public let to: Address
+        public let gas: Int?
+        public let gasPrice: Int?
+        public let value: Int?
+        public let data: String?
+        public let blockParameter: BlockParameter
+        
+        public var method: String {
+            return "eth_call"
+        }
+        
+        public var parameters: Any? {
+            // NOTE: This is to remove key/value if the value is nil.
+            let txParams: [String: Any] = [
+                "from": from?.string as Any?,
+                "to": to.string,
+                "gas": gas,
+                "gasPrice": gasPrice,
+                "value": value,
+                "data": data
+                ].reduce([String: Any]()){ (d, e) in
+                    guard let value = e.value else { return d }
+                    var d = d
+                    d[e.key] = value
+                    return d
+            }
+            return [txParams, blockParameter.rawValue]
+        }
```

Comment:
This part here is a bit confusing to read. please check if the value is nil by `if let` and append element to an array, then return it.

```swift
var params: [Any] = []

if let fromAddress = from?.string {
  params.append(fromAddress)
}

.....

return params
```

Reply:
Ok, I'll think it again!